src/snarl_seed_clusterer.hpp

```diff
@@ -381,59 +453,68 @@ class NewSnarlSeedClusterer {
 
 
-        //Cluster all the snarls at the current level and update the tree_state
-        //to add each of the snarls to the parent level
+        //Cluster all the snarls at the current level
         void cluster_snarl_level(TreeState& tree_state) const;
 
         //Cluster all the chains at the current level
+        //also assigns each chain to its parent and saves the distances to the ends of the parent
+        //for each chain
         void cluster_chain_level(TreeState& tree_state, size_t depth) const;
 
         //Cluster the seeds on the specified node
-        void cluster_one_node(TreeState& tree_state, NodeClusters& node_clusters) const; 
+        //seed_range_start/end are iterators to the range in a vector of seeds for this node
+        //the tuple is <index of node in all_node_clusters, read_num of seed, cluster_num of seed>
+        void cluster_one_node(TreeState& tree_state, NodeClusters& node_clusters,
+                              const std::vector<std::tuple<size_t, size_t, size_t>>::iterator& seed_range_start,
+                              const std::vector<std::tuple<size_t, size_t, size_t>>::iterator& seed_range_end) const; 
 
-        //Cluster the seeds in a snarl given by its net handle
+        //Cluster the seeds in a snarl
         //Snarl_cluster_index is the index into tree_state.all_node_clusters
+        //child_range_start/end are the iterators to the start (inclusive) and end (exclusive) 
+        //of range of the snarl in snarl_to_children
         void cluster_one_snarl(TreeState& tree_state, size_t snarl_clusters_index, 
                  std::multimap<size_t, size_t>::iterator child_range_start, std::multimap<size_t, size_t>::iterator child_range_end) const;
 
-        //Cluster the seeds in a chain given by chain_index_i, an index into
+        //Cluster the seeds in a chain given by chain_clusters_index, an index into
         //distance_index.chain_indexes
-        //If the depth is 0, also incorporate the top-level seeds from tree_state.top_level_seed_clusters
-        //Chain children are tuples<net_handle, (child index, inf) or (seed read num, seed index), chain component, prefix sum>
+        //chain_range_start/end are iterators to the start (inclusive) and end (exclusive) of the
+        //range in chain_to_children representing children of this chain
+        //The range must be ordered
+        //
         //If the children of the chain are only seeds on nodes, then cluster as if it is a node
-        void cluster_one_chain(TreeState& tree_state, size_t chain_clusters_index, vector<tuple<net_handle_t, size_t, size_t, size_t, size_t>>& children_in_chain, bool only_seeds, bool is_top_level_chain) const;
+        void cluster_one_chain(TreeState& tree_state, size_t chain_clusters_index, 
+            const std::vector<ParentToChildMap::ParentChildValues>::iterator& chain_range_start,
+            const std::vector<ParentToChildMap::ParentChildValues>::iterator& chain_range_end,
```

Comment:
I think this indicates that the `ParentToChildMap` isn't actually *for* mapping from a parent to its children. It is laid out organized by parent, but really it's holding scratch for the clustering problem, for all the children, which you are going to visit organized by parent.